### PR TITLE
Added java maven CI

### PR DESCRIPTION
**Pull request by Shernz**
Added actions config to all the branches being pushed or requesting PR

### CODE REVIEW

**Comment by Xnkr** [comment]
Checks are failing because of `pom.xml` which is part of another commit. Wait for it to merge

**Comment by Xnkr** [comment]
You need to rebase with main

**Comment by Shernz** [comment]
Hurrey!!!! 😍🎊

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK 11
+      uses: actions/setup-java@v3
+      with:
+        java-version: '18'
+        distribution: 'temurin'
+        cache: maven
+    - name: Build with Maven
+      run: mvn -B package --file pom.xml
```
Comment (Xnkr) [inline .github/workflows/maven.yml:26]: does `package` also run unit tests?

Comment (Shernz) [reply .github/workflows/maven.yml:26]: yes

**Review by Xnkr — APPROVED** [review]

**Pull request merged**